x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mapping.test.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import '../../../__mocks__/shallow_useeffect.mock';
+import { DEFAULT_INITIAL_APP_DATA } from '../../../../../common/__mocks__';
+import { setMockActions, setMockValues } from '../../../__mocks__';
+import { engines } from '../../__mocks__/engines.mock';
+
+import React from 'react';
+
+import { shallow } from 'enzyme';
+
+import { EuiCheckbox } from '@elastic/eui';
+
+import { Loading } from '../../../shared/loading';
+import {
+  AttributeSelector,
+  DeleteMappingCallout,
+  RoleSelector,
+} from '../../../shared/role_mapping';
+import { asRoleMapping } from '../../../shared/role_mapping/__mocks__/roles';
+
+import { RoleMapping } from './role_mapping';
+
+describe('RoleMapping', () => {
+  const mockRole = DEFAULT_INITIAL_APP_DATA.appSearch.role;
+  const initializeRoleMappings = jest.fn();
+  const initializeRoleMapping = jest.fn();
+  const handleSaveMapping = jest.fn();
+  const handleEngineSelectionChange = jest.fn();
+  const handleAccessAllEnginesChange = jest.fn();
+  const handleAttributeValueChange = jest.fn();
+  const handleAttributeSelectorChange = jest.fn();
+  const handleDeleteMapping = jest.fn();
+  const handleRoleChange = jest.fn();
+  const handleAuthProviderChange = jest.fn();
+  const resetState = jest.fn();
```

Comment:
If I could make a general suggestion - I'd use a collective actions obj, so you can do something like:


```js
const actions = {
  initializeRoleMappings: jest.fn(),
  initializeRoleMapping: jest.fn(),
  // ... etc
};

// allows you to do:

setMockActions(actions); // instead of specifying every single action const
setMockActions({ ...actions, someOverride: 'whatever' }); // easier to override a single fn if needed

// asserting
expect(actions.resetState).toHaveBeenCalled(); // just requires an `action.` prefix
```

Reply:
https://github.com/elastic/kibana/pull/94763/commits/5530cfd4632bec5e5b3f78e62e393b256a3b2b64